RELEASE.md

```diff
@@ -0,0 +1,52 @@
+# Release Process for `connect-go`
+
```

Comment:
I think the list formatting of the notion doc made this more clear because it was a single nested list. It looks like you introduced levels of hierarchy here in this readme that don't exist in the Notion doc. I prefer how the Notion doc is formatted.